### PR TITLE
HLSL support in Visual Studio projects

### PR DESCRIPTION
**Pull request by zoejbare**
- Added support for including HLSL files in generated Visual Studio projects
- Changed the Visual Studio project generator so file type folders are disabled by default

### CODE REVIEW

File: csbuild/tools/project_generators/visual_studio/internal/__init__.py [hunk 1]
```diff
@@ -93,11 +97,15 @@ class Version(object):
...
+	| HLSL_SOURCE_FILE_EXTENSIONS \
+	| HLSL_HEADER_FILE_EXTENSIONS \
 	| ASM_FILE_EXTENSIONS \
+	| PYTHON_FILE_EXTENSIONS \
+	| BATCH_FILE_EXTENSIONS \
 	| MISC_FILE_EXTENSIONS
 
 # Switch for toggling the project folders separating files by their extensions.
-ENABLE_FILE_TYPE_FOLDERS = True
+ENABLE_FILE_TYPE_FOLDERS = False
```
Comment (ShadauxCat) [inline csbuild/tools/project_generators/visual_studio/internal/__init__.py:108]: Why this change?

Comment (zoejbare) [reply csbuild/tools/project_generators/visual_studio/internal/__init__.py:108]: Because while it serves a reasonable purpose for some, most projects disable it.  Made more sense to me to disable it by default.

Comment (zoejbare) [reply csbuild/tools/project_generators/visual_studio/internal/__init__.py:108]: Ideally, this would be set through a command line argument, but we don't have the ability to define custom arguments yet.

**Review by ShadauxCat — APPROVED** [review]
